compiler/semtypinst.nim

```diff
@@ -462,6 +462,14 @@ proc propagateFieldFlags(t: PType, n: PNode) =
       propagateFieldFlags(t, son)
   else: discard
 
+template genericParamsToString(t: untyped): untyped =
```

Comment:
the reasoning `proc` is usually preferred when possible is `template` have some of the same caveats as C macros, including `mytemplate(exprWithSideEffect())` will call `exprWithSideEffect()` twice in `mytemplate(x)=x+x`, and errors are caught earlier with `proc` since its arguments are typed; and `template` can be non-hygienic